chainercv/links/model/fpn/faster_rcnn.py

```diff
@@ -45,18 +48,36 @@ class FasterRCNN(chainer.Chain):
 
     """
 
-    _stride = 32
+    stride = 32
+    _accepted_return_values = ('rois', 'bboxes', 'labels', 'scores', 'masks')
 
-    def __init__(self, extractor, rpn, head,
+    def __init__(self, extractor, rpn, bbox_head,
+                 mask_head, return_values,
                  min_size=800, max_size=1333):
+        for value_name in return_values:
+            if value_name not in self._accepted_return_values:
+                raise ValueError(
+                    '{} is not included in accepted value names {}'.format(
+                        value_name, self._accepted_return_values))
+        self._return_values = return_values
+
+        self._store_rpn_outputs = 'rois' in self._return_values
+        self._run_bbox = any([key in self._return_values
+                              for key in
+                              ['bboxes', 'labels', 'scores', 'masks']])
+        self._run_mask = 'masks' in self._return_values
         super(FasterRCNN, self).__init__()
+
         with self.init_scope():
             self.extractor = extractor
             self.rpn = rpn
-            self.head = head
+            if self._run_bbox:
+                self.bbox_head = bbox_head
+            if self._run_mask:
+                self.mask_head = mask_head
 
-        self._min_size = min_size
-        self._max_size = max_size
+        self.min_size = min_size
+        self.max_size = max_size
```

Comment:
Why did you make these members public?

Reply:
https://github.com/chainer/chainercv/pull/781#discussion_r276521963